### PR TITLE
Add automapper (map User)

### PR DESCRIPTION
**Pull request by PedenkoIllia**
Created map for User <--> UserApiModel
We will create other maps as needed

### CODE REVIEW

File: BusinessLogic/MapperProfile/MapperProfile.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using DataAccess.Context;
+using Domain.EF_Models;
+
+using System.Linq;
+
+namespace TimeOffTracker.WebApi.MapperProfile
+{
+    public class MapperProfile : Profile
+    {
+        private TimeOffTrackerContext _context;
```
Comment (ASbeletsky) [inline BusinessLogic/MapperProfile/MapperProfile.cs]: remove _context from here, mapping shouldn't perform db operations

File: BusinessLogic/MapperProfile/MapperProfile.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        public MapperProfile(TimeOffTrackerContext context)
+        {
+            _context = context;
+            CreateMap<User, UserApiModel>()
+                .ForMember(model => model.Role, opt => opt.MapFrom(usr =>
```
Comment (ASbeletsky) [inline BusinessLogic/MapperProfile/MapperProfile.cs]: It's business logic part. 
The right way to do this

1. Create a UserService class in BusinessLogic/Services/
2. Inject IRepository<User> and IMapper to consctuctor
3. Add Method IEnumeralbe<UserApiModel> GetUsers - mapper.Map<UserApiModel>(repository.GetAll()).ToList()
4. Call GetUsers  in the controller

File: TimeOffTracker.WebApi/Controllers/AccountController.cs [hunk 3]
```diff
@@ -42,7 +48,11 @@ public async Task<HttpStatusCode> Post([FromForm] RegisterViewModel model)
         {
             if (ModelState.IsValid)
             {
-                User user = new User { Email = model.Email, UserName = model.Email };
+                User user = new User
```
Comment (ASbeletsky) [inline TimeOffTracker.WebApi/Controllers/AccountController.cs]: Use Mapper.Map<RegisterViewModel, User>() instead of manual mapping

**Review by ASbeletsky — CHANGES_REQUESTED** [review]

File: TimeOffTracker.WebApi/Controllers/AccountController.cs (hunk 3 above)
Comment (PedenkoIllia) [reply TimeOffTracker.WebApi/Controllers/AccountController.cs]: Done (with turning email into a username)
For this I moved the RegisterViewModel from WEB to ApiModels project.

File: BusinessLogic/MapperProfile/MapperProfile.cs (hunk 1 above)
Comment (PedenkoIllia) [reply BusinessLogic/MapperProfile/MapperProfile.cs]: Done

File: BusinessLogic/MapperProfile/MapperProfile.cs (hunk 2 above)
Comment (PedenkoIllia) [reply BusinessLogic/MapperProfile/MapperProfile.cs]: Done

File: BusinessLogic/Services/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                SetUserRole(usersEnumerator.Current, modelsEnumerator.Current);
+            }
+
+            return userModels;
+        }
+
+        private void SetUserRole(User user, UserApiModel userModel)
+        {
+            userModel.Role = _userManager.GetRolesAsync(user).Result.FirstOrDefault();
```
Comment (ASbeletsky) [inline BusinessLogic/Services/UserService.cs]: Use async await

File: BusinessLogic/Services/UserService.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        public IEnumerable<UserApiModel> GetUsers()
+        {
+            var users = _userManager.Users.ToList();
+            IEnumerable<UserApiModel> userModels = _mapper.Map<IEnumerable<UserApiModel>>(users);
+            
+            var usersEnumerator = users.GetEnumerator();
+            var modelsEnumerator = userModels.GetEnumerator();
+            while (usersEnumerator.MoveNext() && modelsEnumerator.MoveNext())
```
Comment (ASbeletsky) [inline BusinessLogic/Services/UserService.cs]: Use for each, map to user model inside the loop

File: DataAccess/Repository/UserRepository.cs [hunk 6]
```diff
@@ -11,7 +11,7 @@
 
 namespace DataAccess.Repository
 {
-    public class UserRepository: BaseRepository<User, int>
+    public class UserRepository: BaseRepository<User, string>
```
Comment (ASbeletsky) [inline DataAccess/Repository/UserRepository.cs]: Why string?

**Review by ASbeletsky — CHANGES_REQUESTED** [review]

File: DataAccess/Repository/UserRepository.cs (hunk 6 above)
Comment (PedenkoIllia) [reply DataAccess/Repository/UserRepository.cs]: Id in Identity has default type string. We did not change it. The id column is of type varchart(450)

Comment (ASbeletsky) [reply DataAccess/Repository/UserRepository.cs]: should be int, autoincremented identity

**Review by ASbeletsky — APPROVED** [review]

**Pull request merged**